util/misc.go

```diff
@@ -531,3 +543,62 @@ func GetLocalIP() string {
 	}
 	return ""
 }
+
+func createTLSCertificates() error {
+
+	privkey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return err
+	}
+
+	notBefore := time.Now()
+	notAfter := notBefore.Add(90 * 24 * time.Hour)
```

Comment:
Should these 90 days be configurable? Or just documented that if you don't manage any TLS certificates, there will be a temporary self-signed cert created that is valid for 90 days and then you need to restart the TiDB node?

Reply:
Good question. Not sure what to do. For now I've added the validity of the cert to the logging, however `zap.Duration` doesn't format it in the best way possible...

We could:
1. Make the certificates valid for 3 years and expect people to restart before they reach this. Should this be 1 year? 10 years?
2. We could re-create the certs every 60 days or so and trigger the code behind `ALTER INSTANCE RELOAD TLS`
3. We could put a note in the docs saying that the user has to restart every <90 days

This is related to https://github.com/pingcap/docs/issues/4299

I think option 2 is probably best. What do you think?

Could/should we have a hook or so that we call when the certificates are about to expire? Then this allows integration with ACME and local certificate rotation things. But I also don't want to make this too complicated.

Reply:
https://github.com/pingcap/tidb/blob/59b99ce1d4de338867a5f98c9d23ad1d6b3703e5/executor/simple.go#L1354-L1372 is probably close to what we need.

Reply:
I've updated the code to have a goroutine create new certs every 30 days.